v2/lints/rfc/lint_tbs_signature_alg_matches_cert_signature_alg.go

```diff
@@ -0,0 +1,92 @@
+package rfc
+
+import (
+	"bytes"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/util"
+	"golang.org/x/crypto/cryptobyte"
+	cryptobyte_asn1 "golang.org/x/crypto/cryptobyte/asn1"
+)
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*******************************************************************
+RFC 5280: 4.1.1.2
+[the Certificate signatureAlgorithm] field MUST contain the same
+algorithm identifier as the signature field in the sequence
+tbsCertificate
+********************************************************************/
+
+type mismatchingSigAlg struct{}
+
+func (l *mismatchingSigAlg) Initialize() error {
+	return nil
+}
+
+func (l *mismatchingSigAlg) CheckApplies(_ *x509.Certificate) bool {
+	return true
+}
+
+func (l *mismatchingSigAlg) Execute(c *x509.Certificate) *lint.LintResult {
+	// parse out certificate signatureAlgorithm
+	input := cryptobyte.String(c.Raw)
+	var cert cryptobyte.String
+	if !input.ReadASN1(&cert, cryptobyte_asn1.SEQUENCE) {
+		return &lint.LintResult{Status: lint.Fatal, Details: "error reading certificate"}
+	}
+	if !cert.SkipASN1(cryptobyte_asn1.SEQUENCE) {
+		return &lint.LintResult{Status: lint.Fatal, Details: "error reading certificate.tbsCertificate"}
+	}
+	var certSigAlg cryptobyte.String
+	if !cert.ReadASN1(&certSigAlg, cryptobyte_asn1.SEQUENCE) {
+		return &lint.LintResult{Status: lint.Fatal, Details: "error reading certificate.signatureAlgorithm"}
+	}
+
+	// parse out tbsCertificate signature
+	input = cryptobyte.String(c.RawTBSCertificate)
+	var tbsCert cryptobyte.String
+	if !input.ReadASN1(&tbsCert, cryptobyte_asn1.SEQUENCE) {
+		return &lint.LintResult{Status: lint.Fatal, Details: "error reading tbsCertificate"}
+	}
+	if !tbsCert.SkipOptionalASN1(cryptobyte_asn1.Tag(0).Constructed().ContextSpecific()) {
```

Comment:
small nit: maybe worth pulling out `cryptobyte_asn1.Tag(0).Constructed().ContextSpecific()` as an unexported package level `var` to avoid needing to construct it each time?

Reply:
I think after another bit of thought it would be less readable.